File: calendar.java
import java.util.Scanner;

public class calendar{
    //Declaring global entities to be used
    static Scanner in= new Scanner(System.in);
    static int month=1,d=0,r=0,day=6;
    static long rep=1,year=0;
    //Array with month names to be displayed on the top of every months array in the calendar
    static String m[]= {"January","February","March","April","May","June","July","August","September","October","November","December"};
    static String mat[][]= new String[6][7];
    //Function to clear the array for next month
    public void clear(){
      display();
      mat= new String[6][7];
      r=0;
      d=0;
      month++;}
    //Function to display the current month values in an array
    public void display(){
     if(rep==year){
     System.out.println();//Decorating the calendar with some basic lines
     System.out.println("      "+m[month-1]);
     System.out.println("--------------------");
     System.out.println("Su|Mo|Tu|We|Th|Fr|Sa");
     System.out.println("--------------------");
     for(int i=0;i<6;i++){
      for(int j=0;j<7;j++){
       if(mat[i][j]==null)//For those dates which starts from middle of week
        System.out.print("   ");
       else
        System.out.print(mat[i][j]+" ");}
        System.out.println();}}}

    public static void main(String args[]){
        calendar ob= new calendar();//To call the above functions
        System.out.print("Enter the year in numeric to get its calendar: ");
        year= in.nextLong();
        while(rep<=year){//Repetition till the process reach the entered year for its calendar 
            month= 1;
            while(month<=12){//Repetition of array filling till next year starts rep=rep+1
            d++;
            if((d==3) && (month==9) && (rep==1752)){ //For the missing 11 days from the past calendars
            d=14;
            day=4;}
            if(d<10)//Converting single digit dates to double digit
            mat[r][day]= "0"+d;
            else
            mat[r][day]= d+"";
            day++;
            if(day>6){//Logic to start the new week day=0
            day=0;
            r++;}
            //Changing the month if previous one was of 31 days
            if((d==31)&&((month==1)||(month==3)||(month==5)||(month==7)||(month==8)||(month==10)))
             ob.clear();
            //Changing the month if previous one was of 30 days
            else if((d==30)&&((month==4)||(month==6)||(month==9)||(month==11)))
              ob.clear();
            //Changing the month if previous one was of February
            else if(month==2){
               //Logic for non-leap years containing 28 days in February and 365 days in the year
               if((d==28)&&(((rep%100!=0)&&(rep%4!=0))||((rep%100==0)&&((rep>1700)&&(rep%400!=0)))))
                ob.clear();
               //Logic for leap years containing 29 days in February and 366 days in the year
               else if((d==29)&&(((rep%100!=0)&&(rep%4==0))||((rep%100==0)&&(rep%400==0)&&(rep>1700))||((rep%100==0)&&(rep<=1700))))
                ob.clear();}
            //Changing the month to January in the beginning of the outer loop's body and year here
            else if((month==12)&&(d==31)){
                ob.clear();
                //Increase in rep means changing the year from one ahead to continue the process
                rep++;}}}}}

File: Calendar.py
#Initializing global variables for creating calendar systematically
month=1
d=0
day=6
rep=1
#list containing names of all the months to be displayed on the calendar
m= ["January","February","March","April","May","June","July","August","September","October","November","December"]
a=list() #list to store calendar of a month
b=list() #list to store calendar of a week

def reset(): #function to clear list to generate calendar of next month or new year's month
   #calling global variables in the function
   global rep, day, d, month, m, a, b
   a=list()
   d=0
   month= month+1 #moving to the next month

def display(year): #function to display the calendar of a month
    #calling global variables in the function
    global rep, day, d, month, m, a, b
    a.append(b) #filling the calendar of last-halved week inside the list of month: b=list()
    if(rep==year): #printing the calendar of the entered year
     #designing the calendar using strings and new lines
     print()
     print("      "+m[month-1]) #displaying the name of month just before it's calendar
     print("--------------------")
     print("Su|Mo|Tu|We|Th|Fr|Sa") #displaying the name of days in a week
     print("--------------------")
     #printing the calendar of a month
     for i in a:
        for j in i:
           if(j==" "):
            print("   ",end= "")
           else:
             print(j+" ",end="")
        print() #changing lines of dates with a new week

def main(): #main function to generate calendar's
 #calling global variables
 global rep, day, d, month, m, a, b
 year= int(input("Enter a year in numeric to get its calendar: ")) #taking input of a year 
 while(rep<=year):                                                 #to display it's calendar
   month= 1  #changing the month to january=1 with change in year in process
   while(month<=12): 
    if(d==0):
      b= list() #clearing the last halved week from the list of previous month
      for i in range(day):
        b.append(" ") #filling the first week with blank dates
    d= d+1 #changing the dates in a month
    if((d==3) and (month==9) and (rep==1752)): #Skipping 11 days for year 1752 in history
      d=14
      day=4
    if(d<10): #converting dates 0 to 9 to double digit form
      b.append("0"+str(d))
    else:
      b.append(str(d))
    day= day+1 #changing days with the date
    if(day>6): #changing the week starting from sunday: day=0
       day=0
       a.append(b) #filling a complete week in list of month
       b= list() #clearing the list of week for dates of next week
    # Logic to change month for the month containing 31 days
    if((d==31)and((month==1)or(month==3)or(month==5)or(month==7)or(month==8)or(month==10))):
     display(year)
     reset()
    # Logic to change month for the month containing 30 days
    elif((d==30)and((month==4)or(month==6)or(month==9)or(month==11))):
     display(year)
     reset()
     # Logic to change month if the current month is February
    elif(month==2):
      #Logic for non-leap year February containg 28 days
      if((d==28)and(((rep%100!=0)and(rep%4!=0))or((rep%100==0)and((rep>1700)and(rep%400!=0))))):
       display(year)
       reset()
      #Logic for leap year February containg 29 days
      elif((d==29)and(((rep%100!=0)and(rep%4==0))or((rep%100==0)and(rep%400==0)and(rep>1700))or((rep%100==0)and(rep<=1700)))):
       display(year)
       reset()
    # Logic to change year with reseting the month to january: month=1 if the current month is December
    elif((month==12)and(d==31)):
       display(year)
       reset()
       rep= rep+1

if __name__=="__main__": # Running the main function as initial
 main()